[Rehman Sajid] iP

rehmmann opened the pull request:
# Introducing Duke

> "It always seems impossible until it is done" - Nelson Mandela

## Why use it?
- Runs smoothly
- Made with your needs at heart
- *Most importantly it's **Free***

## Features
1. Store events, deadlines, todos
2. Writes & loads from hard-disk 😄 


Some basic Duke commands are:
```
todo task1
event task2/12-12-2010
deadline task3/12-02-2010
list
bye
```

Credits to : [Favourite module CS2103](https://nus-cs2103-ay2122s1.github.io/website/index.html)

Commit:
Add Gradle support

rehmmann committed:
Level 1

rehmmann committed:
Level 2

rehmmann committed:
Used classes to complete Level-3

rehmmann committed:
created subClasses & completed Level-4

rehmmann committed:
Automated testing

rehmmann committed:
Errors handled

rehmmann committed:
Chatbot able to delete tasks

rehmmann committed:
Custom exception file added

rehmmann committed:
Save and Read items from list

rehmmann committed:
Include date-time functionality

rehmmann committed:
Resolve merge confilicts

rehmmann committed:
Refactor Parser Ui Storage TaskList classes

rehmmann committed:
Create duke package

rehmmann committed:
Test cases with JUNIT

rehmmann committed:
Create JAR

rehmmann committed:
Add JavaDocs

rehmmann committed:
Follow coding standards

rehmmann committed:
Include search functionality

rehmmann committed:
Merge branch 'branch-A-CodingStandard'

rehmmann committed:
Fix merge conflicts

rehmmann committed:
Fix indentations

rehmmann committed:
Merge branch 'add-gradle-support' of https://github.com/rehmmann/ip into add-gradle-support

rehmmann committed:
Set up gradle

rehmmann committed:
Use checkstyle

hernpiblo commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+
+class Deadline extends Task {//Need to be done by deadline
```
Should the comments be made as a Header comment for the class (Javadocs)

rehmmann replied:
Yeap thanks for pointing it out!

hernpiblo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+class Duke {
+
+    Scanner myScanner = new Scanner(System.in);
+
```
Theres extra lines/whitespaces here which could affect readability

hernpiblo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String currLine = s.nextLine();
+//            System.out.println(currLine);
+            char taskType = currLine.charAt(1);
+//            System.out.println(taskType);
+            switch (taskType) {
+                case 'T':
+                    toDo.add(new ToDo(currLine.substring(7)));
+                    System.out.println(new ToDo(currLine.substring(7)));
+                    break;
+                case 'D':
+                    int l = currLine.indexOf("(");
+//                    int m = currLine.indexOf("by ");
+                    int n = currLine.indexOf(")");
+//                    System.out.println(currLine.substring(7,l));
+//                    System.out.println(currLine.substring(l+1,n));
+                    toDo.add(new Deadline(currLine.substring(7, l), currLine.substring(l + 1, n)));
+                    System.out.println(new Deadline(currLine.substring(7, l), currLine.substring(l + 1, n)));
+                    break;
+                case 'E':
+                    int i = currLine.indexOf("(");
+//                    int j = currLine.indexOf("at ");
+                    int k = currLine.indexOf(")");
+//                    System.out.println(currLine.substring(7,i));
+//                    System.out.println(currLine.substring(j,k));
```
I think commented out code can be deleted to improve readability

hernpiblo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+            char taskType = currLine.charAt(1);
+//            System.out.println(taskType);
+            switch (taskType) {
+                case 'T':
```
i believe case statements should not be indented according to our modules coding standards :)

hernpiblo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+    Storage storage;
+    TaskList tasklist;
+    Scanner myScanner;
+    boolean breakLoop = false;
```
Perhaps boolean variable names can be renamed to sound like boolean

hernpiblo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+        breakLoop = true;
+    }
+
+    /**
+     * Calls appropriate UI functions based on input
+     *
+     * @param input String passed by user
+     */
```
I like how theres javadocs for this public method :)

hernpiblo reviewed (commented):
Overall looks good! some issues include having a number of commented out lines of code which can be deleted to make the code neater. And having header javadocs for public methods instead of inline commenting beside the method.

Looks good nonetheless!

rehmmann committed:
Added basic GUI

rehmmann committed:
Add assert statements

These insertions would minimise bugs in future

rehmmann committed:
Merge pull request #1 from rehmmann/branch-A-Assertions

Add assert statements

rehmmann committed:
Clean up files

rehmmann committed:
Merge pull request #2 from rehmmann/branch-A-CodeQuality

Improve code quality

rehmmann committed:
Modify Parser & UI classes

rehmmann committed:
Merge branch 'master' of https://github.com/rehmmann/ip

rehmmann committed:
Statistics extension
Currently we donot have any stats for users
We want users to be able to see the distribution of their tasks
Now they can see how many tasks they have of each type

rehmmann committed:
Include progress tracker

rehmmann committed:
Remove CI

rehmmann committed:
Improve Ui

rehmmann committed:
Include Gui sample

rehmmann committed:
Add basic user guide

rehmmann committed:
Set theme jekyll-theme-slate

rehmmann committed:
Fix checkstyle issues in test files

rehmmann committed:
Merge branch 'master' of https://github.com/rehmmann/ip

rehmmann committed:
Fix minor checkstyle issues

rehmmann committed:
Javadocs added

rehmmann committed:
Update README.md

rehmmann committed:
Resolve some issues raised in peer evaluation

rehmmann committed:
Resolve merge conflicts

rehmmann committed:
Modify ReadMe

rehmmann committed:
Fix header comment issues

rehmmann committed:
Update README.md

rehmmann committed:
Update README.md
